superset-frontend/src/components/MetadataBar/MetadataBar.test.tsx

```diff
@@ -265,3 +265,27 @@ test('correctly renders the tags tooltip', async () => {
     );
   });
 });
+
+test('renders StyledItem with role="button" when onClick is defined', () => {
+  const onClick = jest.fn();
+  const items = [
+    { ...ITEMS[0], onClick },
+    { ...ITEMS[1], onClick },
+  ];
+  render(<MetadataBar items={items} />);
+
+  const styledItems = screen.getAllByRole('button');
+
+  styledItems.forEach(styledItem => {
+    expect(styledItem).toHaveAttribute('role', 'button');
+  });
+});
+
+test('renders StyledItem with role=undefined when onClick is not defined', () => {
+  const items = [ITEMS[0], ITEMS[1]];
+  render(<MetadataBar items={items} />);
+  const styledItem = screen
+    .getByText(DASHBOARD_TITLE)
+    .closest('.metadata-text');
+  expect(styledItem).not.toHaveAttribute('role');
```

Comment:
Maybe we can do the opposite here and verify that the length of elements with `role=button` is zero?

Reply:
This will be changed to: 
```
  const styledItems = screen.queryAllByRole('button');

  expect(styledItems.length).toBe(0);
```